radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java

```diff
@@ -88,29 +88,30 @@
 		private final ConnectableObservable<AtomSubmissionUpdate> updates;
 		private final Completable completable;
 
-		private Result(ConnectableObservable<AtomSubmissionUpdate> updates) {
+		private Result(ConnectableObservable<AtomSubmissionUpdate> updates, Completable completable) {
 			this.updates = updates;
-
-			this.completable = updates.filter(AtomSubmissionUpdate::isComplete)
-				.firstOrError()
-				.flatMapCompletable(update -> {
-					if (update.getState() == AtomSubmissionState.STORED) {
-						return Completable.complete();
-					} else {
-						return Completable.error(new ActionExecutionException(update.getData().toString()));
-					}
-				});
+			this.completable = completable;
 		}
 
 		private Result connect() {
 			this.updates.connect();
 			return this;
 		}
 
+		/**
+		 * A low level interface, returns an a observable of the status of an atom submission as it occurs.
```

Comment:
Should it be documented that this is a ConnectableObservable underneath and someone needs to connect to it first?

Reply:
The logic is encapsulated privately in RadixApplicationAPI so I think it should be okay.